src/Transit/Internal/Network.hs

```diff
@@ -111,32 +114,25 @@ getSocketPort (Just sock') = do
   portnum <- socketPort sock'
   return (Just portnum)
 
-type Hostname = Text
-
-ipv4ToHostname :: Word32 -> Hostname
-ipv4ToHostname ip =
-  let (q1, r1) = ip `divMod` 256
-      (q2, r2) = q1 `divMod` 256
-      (q3, r3) = q2 `divMod` 256
-  in
-    show r1 <> "." <> show r2 <> "." <> show r3 <> "." <> show q3
-
 buildDirectHints :: PortNumber -> IO (Set.Set ConnectionHint)
 buildDirectHints portnum = do
   nwInterfaces <- getNetworkInterfaces
   let nonLoopbackInterfaces =
-        filter (\nwInterface ->
-                   let (IPv4 addr4) = ipv4 nwInterface
-                   in
-                     (ipv4ToHostname addr4 /= "0.0.0.0")
-                     && (ipv4ToHostname addr4 /= "127.0.0.1"))
-        nwInterfaces
-  return $ Set.fromList $ map (\nwInterface ->
-                                 let (IPv4 addr4) = ipv4 nwInterface in
-                                   Direct Hint { hostname = ipv4ToHostname addr4
-                                               , port = fromIntegral portnum
-                                               , priority = 0
-                                               , ctype = DirectTcpV1 }) nonLoopbackInterfaces
+          filter (\nwInterface -> name nwInterface /= "lo" &&
```

Comment:
Alas, there can be multiple loopback interfaces with different names (lo0, lo1, etc, are common).

Also!  They don't even have to have the address "127.0.0.1" - at least on Linux.  Linux considers 127.0.0.0/8 loopback.  So the old version is also probably not entirely correct.

It's probably okay to punt this to a future issue.  Could you file one and link to it from here?